sentinel-core/src/main/java/com/alibaba/csp/sentinel/util/ConfigUtil.java

```diff
@@ -107,20 +112,11 @@ private static Properties loadPropertiesFromClasspathFile(String fileName) {
 
         Properties properties = new Properties();
         for (URL url : list) {
-            try {
+            try (BufferedReader bufferedReader =
+                         new BufferedReader(new InputStreamReader(url.openStream(), getCharset().newDecoder()))) {
```

Comment:
What's the difference between `InputStreamReader(url.openStream(), getCharset().newDecoder())` and `InputStreamReader(url.openStream(), getCharset())`? It seems that the latter one is equivalent to:

```java
new InputStreamReader(url.openStream(),
  getCharset().newDecoder()
     .onMalformedInput(CodingErrorAction.REPLACE)
     .onUnmappableCharacter(CodingErrorAction.REPLACE))
```

Reply:
When `malformedInputAction` or `unmappableCharacterAction` occurs: 
- `InputStreamReader(url.openStream(), getCharset())` will replaced by a special character. 
- `InputStreamReader(url.openStream(), getCharset().newDecoder())` will throws  `MalformedInputException` or `UnmappableCharacterException`

Reply:
Both have pros and cons. The `REPLACE` action is commonly used, as it won't affect normal attributes.

Reply:
Yes, this way you can continue reading data even if something goes wrong.

Reply:
So here we may change to `InputStreamReader(url.openStream(), getCharset())`?

Reply:
How do we let the user know that the read property is wrong, without throwing an exception. 

The unknown error attribute is a hidden bomb.

Reply:
`ConfigUtil.java` may be just a reading tool, Its job is to reading the contents of the file。The correctness of the contents of the file should be guaranteed by the file provider itself, not the reading tool. Later, consider presenting the current configuration on the console and supporting dynamic editing.
 

Reply:
> How do we let the user know that the read property is wrong, without throwing an exception.

The malformed attribute might be discovered in the log.

Reply:
Thanks guys, I will change it later.